File: src/main/java/com/intetics/lukyanenko/models/Order.java
package com.intetics.lukyanenko.models;

import java.util.Date;
import java.util.List;

public class Order
        extends Common
{
  private int               id;
  private String            number;
  private Date              created;
  private boolean           isBasket;
  private Customer          customer;
  private Integer           customerId;
  private List<OrderDetail> details;
  
  public int getId()
  {
    return id;
  }
  
  public void setId(int id)
  {
    this.id = id;
  }
  
  public String getNumber()
  {
    return number;
  }
  
  public void setNumber(String number)
  {
    this.number = number;
  }
  
  public Date getCreated()
  {
    return created;
  }
  
  public void setCreated(Date created)
  {
    this.created = created;
  }
  
  public boolean getIsBasket()
  {
    return isBasket;
  }
  
  public void setIsBasket(boolean isBasket)
  {
    this.isBasket = isBasket;
  }
  
  public Integer getCustomerId()
  {
    return customerId;
  }
  
  public void setCustomerId(int customerId)
  {
    this.customerId = customerId;
    if (customer != null && customer.getId() != customerId)
      customer = null;
  }
  
  public List<OrderDetail> getDetails()
  {
    return details;
  }
  
  public void setDetails(List<OrderDetail> details)
  {
    this.details = details;
  }
  
  public Double getTotal()
  {
    if (details != null)
    {
      double total = 0.0;
      for(OrderDetail row: details)
        total += row.getQuantity() * row.getItemPrice();
      return total;
    }
    else
      return null;
  }
  
  public Customer getCustomer()
  {
    if (customerId == null)
      return null;
    else if (customer == null)
      customer = daoFactory.getDAO(Customer.class).getByID(customerId);
    return customer;
  }
  
  public void setCustomer(Customer customer)
  {
    this.customer = customer;
    this.customerId = customer.getId();
  }
  
}

File: src/main/java/com/intetics/lukyanenko/models/Order.kt
package com.intetics.lukyanenko.models

import java.util.*

class Order : Common() {
    var id: Int = 0
    var number: String? = null
    lateinit var created: Date
    var isBasket: Boolean = false
    var customer: Customer? = null
        get() =
            field ?: customerId?.let {
                field = daoFactory.getDAO(Customer::class).getByID(it)
                field
            }
        set(value) {
            field = value
            customerId = value?.id
        }

    var customerId: Int? = null
        set(value) {
            if (customer?.id != value)
                customer = null
            field = value
        }

    var details: MutableList<OrderDetail>? = null

    fun getTotal(): Double? = details?.sumByDouble { it.quantity * it.itemPrice } // преимущество Kotlin от Java стоит показать на этом примере
}